shard/shard_state.go

```diff
@@ -382,23 +385,52 @@ func (c *Committee) DeepCopy() Committee {
 	return r
 }
 
+// Hash ..
+func (c *Committee) Hash() common.Hash {
+	return hash.FromRLPNew256(c)
+}
+
+var (
+	blsKeyCache singleflight.Group
+)
+
+func lookupBLSPublicKeys(
+	c *Committee,
+) ([]*bls.PublicKey, error) {
+	key := c.Hash().Hex()
+	results, err, _ := blsKeyCache.Do(
+		key, func() (interface{}, error) {
+			slice := make([]*bls.PublicKey, len(c.Slots))
+			for j := range c.Slots {
+				committerKey := &bls.PublicKey{}
+				if err := c.Slots[j].BlsPublicKey.ToLibBLSPublicKey(
+					committerKey,
+				); err != nil {
+					return nil, err
+				}
+				slice[j] = committerKey
+			}
+			// Only made once
+			go func() {
+				time.Sleep(25 * time.Minute)
```

Comment:
why 25 minutes hardcoded?

Reply:
it's just a reasonable time to keep the cached result.

Reply:
opened a ticket to keep track of this issue which has to be fixed later.

Reply:
https://github.com/harmony-one/harmony/issues/2559